### PR TITLE
enhancement: JSON variable file support

### LINKED ISSUE #4532: JSON variable file support
### Problem

`robot --variablefile burek.json -i smoke ROBOT/` returns the following error:
`failed: Importing variable file ’burek.json’ failed: Not a valid file or directory to import.`

while `burek.py` containing the same JSON content seem to work just fine. 

`--variablefile (-V)` only seems to support Python and YAML file types. 
We would like to use JSON file as a variablefile to pass on certain variables that are used by other frameworks as well.

Thanks for all the help!

### PR DESCRIPTION
**Pull request by sunday2**
Closes #4532

### CODE REVIEW

**Review by pekkaklarck — CHANGES_REQUESTED** [review]
Thanks for a great PR and sorry for reviewing taking so long. Only real issue I noticed in code and tests was that the extension wasn't case-insensitive as it should be, but I added also some other comments. Another thing to do is writing documentation. It can be done after merging this PR as well.

**Comment by sunday2** [comment]
> Thanks for a great PR and sorry for reviewing taking so long. Only real issue I noticed in code and tests was that the extension wasn't case-insensitive as it should be, but I added also some other comments. Another thing to do is writing documentation. It can be done after merging this PR as well.

**Comment by sunday2** [comment]
> Thanks for a great PR and sorry for reviewing taking so long. Only real issue I noticed in code and tests was that the extension wasn't case-insensitive as it should be, but I added also some other comments. Another thing to do is writing documentation. It can be done after merging this PR as well.

File: src/robot/variables/filesetter.py [hunk 1]
```diff
@@ -147,3 +150,27 @@ def _validate(self, name, value):
...
+class JsonImporter:
+    def import_variables(self, path, args=None):
+        if args:
+            raise DataError('JSON variable files do not accept arguments.')
+        variables = self._import(path)
+        return [('${%s}' % name, self._dot_dict(value))
+                for name, value in variables]
+
+    def _import(self, path):
+        with io.open(path, encoding='UTF-8') as stream:
```
[the remark this answers is not in the document]
Comment (sunday2) [reply src/robot/variables/filesetter.py:164]: Thanks for the explain and I will not change both of them in this PR.

**Comment by sunday2** [comment]
> 

Thank you for the review and all the comments, I have updated the code. If there is any other concern, pls don't hesitate to let me know. For writing document, maybe I need to spend some time on studying how to write it.

**Comment by sunday2** [comment]
> Thanks for a great PR and sorry for reviewing taking so long. Only real issue I noticed in code and tests was that the extension wasn't case-insensitive as it should be, but I added also some other comments. Another thing to do is writing documentation. It can be done after merging this PR as well.

Thank you for the review and all the comments, I have updated the code. If there is any other concern, pls don't hesitate to let me know. For writing document, maybe I need to spend some time on studying how to write it.

image

**Comment by pekkaklarck** [comment]
Thanks for a great PR! I can write the documentation myself, I think it can mainly refer to existing YAML variable file documentation so it's a small task.

**Pull request merged**